src/almanack/entropy.py

```diff
@@ -0,0 +1,39 @@
+"""
+This module finds the amount of entropy on a file based level
+"""
+
+import math
+
+from .git_parser import calculate_loc_changes
+
+
+def calculate_shannon_entropy(
+    repo_path: str, source_commit: str, target_commit: str, file_names: list[str]
+) -> dict[str, float]:
+    """
+    Calculates the Shannon entropy for changes in specified files between two commits.
+
+    Args:
+        repo_path (str): The path to the git repository.
```

Comment:
Suggesting specificity here to help distinguish from repo URL's.
```suggestion
        repo_path (str): The file path to the git repository.
```